### ISSUE TITLE
HM.Layout.Order.WrongOrder false positives

### ISSUE THREAD

**Issue by peterwilsoncc** [humanmade/coding-standards#5]
`HM.Layout.Order.WrongOrder` can throw a false positives in files with anonymous functions using use.

This code 

```php
<?php
namespace Hamilton;

use Hamilton\Mixtape;

const WRITER = 'Lin-Manuel Miranda';

add_filter( 'Hamilton/Cast', function ( $cast ) use ( $replacements ) {
	return $replacements;
} );
```

Error thrown:

```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 1 | ERROR | use found on line 8, but const was declared on line 6.
   |       | Statements should be ordered `namespace`, `use`, `const`,
   |       | `require`, then code. (HM.Layout.Order.WrongOrder)
----------------------------------------------------------------------
```

**Comment by dan-westall** [comment]
Yep i'm seeing this aswell, I've disabled the rule for now.

**Comment by connerbw** [comment]
In `\HM\Sniffs\Layout\OrderSniff` add:

```php
// Closures can inherit variables from the parent scope with the `use` language construct. Skip these.
if ( $next_token['code'] == T_USE && $phpcsFile->findPrevious( T_CLOSURE, $next_pos, null, false, null, $local = true ) !== false ) {
    continue;
}
```

I will submit a pull request.
